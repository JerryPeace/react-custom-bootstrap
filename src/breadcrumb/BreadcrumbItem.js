import PropTypes from 'prop-types';
import React from 'react';
import classNames from 'classnames';
import bootstrapUtils from '../utils/bootstrapUtils';

class BreadcrumbItem extends React.Component {

    static propTypes = {
        active: PropTypes.bool
    };

    static defaultProps = {
        active: false
    };

    constructor (props) {
        super(props);
    }

    render() {
        const {active, children, ...rest} = this.props;
        // add specific classes
        let classes = {
            'active': active
        };

        let bsCls = bootstrapUtils.getCls(this.props);

        if (active) {
            return (
                <li {...rest} className={classNames(this.props.className, classes, bsCls)}>
                    {children}
                </li>
            );
        } else {
            return (
                <li {...rest} className={classNames(this.props.className, classes)}>
                    {children}
                </li>
            );
        }
    }

}

export default BreadcrumbItem;
