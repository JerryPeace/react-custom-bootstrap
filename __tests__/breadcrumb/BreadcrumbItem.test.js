import React from 'react';
import { mount } from 'enzyme';
import toJSON from'enzyme-to-json';
import BreadcrumbItem from 'breadcrumb/BreadcrumbItem';

describe('BreadcrumbItem', () => {
    test('default render', () => {
        const breadcrumb = mount(
            <BreadcrumbItem>
                <div>test</div>
            </BreadcrumbItem>
        );
        expect(toJSON(breadcrumb)).toMatchSnapshot();
    });

    test('apply bootstrap style', () => {
        const breadcrumb = mount(
            <BreadcrumbItem active={true} bsClass={'banner'} bsStyle={'info'}>
                <div>test</div>
            </BreadcrumbItem>
        );
        expect(toJSON(breadcrumb)).toMatchSnapshot();
    });
});
