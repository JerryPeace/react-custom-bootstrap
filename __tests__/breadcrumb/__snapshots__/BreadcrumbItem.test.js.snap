// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadcrumbItem apply bootstrap style 1`] = `
<BreadcrumbItem
  active={true}
  bsClass="banner"
  bsStyle="info"
>
  <li
    bsClass="banner"
    bsStyle="info"
    className="active banner banner-info"
  >
    <div>
      test
    </div>
  </li>
</BreadcrumbItem>
`;

exports[`BreadcrumbItem default render 1`] = `
<BreadcrumbItem
  active={false}
>
  <li
    className=""
  >
    <div>
      test
    </div>
  </li>
</BreadcrumbItem>
`;
